packages/ppx/src/css_to_emotion.re

```diff
@@ -220,18 +205,51 @@ and render_media_query = (at_rule: at_rule): Parsetree.expression => {
     );
   };
 }
+and render_component_value = (value: component_value) => {
+  // TODO: More tweaking to be done!
+  switch (value) {
+  | Variable(variable) => render_variable_as_string(variable)
+  | Bracket_block(values) => render_component_values(values)
+  | Paren_block(values) => render_component_values(values)
+  // x%
+  | Percentage(percent) => percent ++ "%"
+  // right bottom
+  | Ident(ident) => ident ++ " "
+  | String(str) => str
+  | Selector(selectors) => render_selectors(selectors)
+  | Uri(uri) => uri
+  | Operator(op) => op
+  // a + b
+  | Combinator(c) => " " ++ c ++ " "
+  // z - y
+  | Delim(delim) => delim ++ " "
+  // rgb( ... )
+  | Function(str_with_loc, values_with_loc) =>
+    let (fn, _) = str_with_loc;
+    let (values, _) = values_with_loc;
+    fn ++ "(" ++ render_component_values(values) ++ ")";
+  // #v
+  | Hash(v) => "#" ++ v
+  // 1 2 3
+  | Number(num) => num ++ " "
+  | Unicode_range(code) => code
+  // 30px or 20px 10px
+  | Float_dimension((dim, unit)) => dim ++ unit ++ " "
```

Comment:
Because of this change, there are some test diffs. That is:
```
   Expected: `"((CssJs.style [|((CssJs.media {js|(min-width: 33px)|js} [||])[@bs ])|])\n[@bs ])"'

   Received: `"((CssJs.style [|((CssJs.media {js|(min-width :   33px )|js} [||])[@bs ])|])\n[@bs ])"'
```
The extra spaces in the received result are because of the string `" "` which particularly takes care of cases likes
`2px 3px`, without it, the latter would be rendered as `2px3px`.

@davesnx, should we ignore that diff or it's sensitive?

Reply:
In those cases we need to capture WS+, so many WS would be only counted one

Reply:
That sounds right!

Reply:
> In those cases we need to capture WS+, so many WS would be only counted one

This seems to capture the `ws + ws` between `:` and `33px` in the example above, but not one (`ws`) before `:` or after `..px`.

Reply:
cases like `... ++ " "` and `" " ++ ...` are the reason for the added whitespace. The lexer handles these right, except `render_component_values` which happens to render strings and so assumes whitespace is not a component value, because of that assumption, I found a need to explicitly add the whitespace in form of `" "`.

Reply:
In `value_in_prelude` production there's this distinction where we care about whitespace: https://github.com//davesnx/styled-ppx/blob/c0deb4ef0b51ca128759ee43b339011f39b7ee92/packages/parser/lib/css_parser.mly#L514